### ISSUE TITLE
Nest backend: default synapse delay conflicts with setting of max_delay to less then 1.0

### ISSUE THREAD

**Issue by antolikjan** [NeuralEnsemble/PyNN#472]
Default delay on StaticSynapse is 1.0 which causes conflicts when max_delay of Nest simulator is set to less then 1.0. This is presumably because also recorders use delay, so even if user sets all other 'real' connections delays in the model to be between the min_delay and max_delay, the nest simulator still complains about synapses with delay 1.0, i.e.:
```
pynestkernel.NESTError: BadDelay in Connect_g_g_D_D: Delay value 1 is invalid: Default delay of 'static_synapse' must be between min_delay 0.1 and max_delay 0.3.
```

**Comment by apdavison** [comment]
Minimal test case to reproduce:
```python
import pyNN.nest as sim
sim.setup(min_delay=0.1, max_delay=0.3)
p = sim.Population(10, sim.IF_cond_exp())
p.record(['spikes', 'v'])
sim.run(10.0)
```

**Comment by apdavison** [comment]
Fixed in 62e58a1fff4c85fcd2757a372d8fcc9335c1b95f
